packages/wrangler/src/bundle.ts

```diff
@@ -779,17 +789,51 @@ async function applyFirstPartyWorkerDevFacade(
  * This code be removed from here when the API is in Workers core,
  * but moved inside Miniflare for simulating D1.
  */
-
 async function applyD1BetaFacade(
 	entry: Entry,
 	tmpDirPath: string,
 	betaD1Shims: string[],
-	local: boolean
+	local: boolean,
+	doBindings: DurableObjectBindings
 ): Promise<Entry> {
+	let entrypointPath = path.resolve(
+		getBasePath(),
+		"templates/d1-beta-facade.js"
+	);
+	if (doBindings.length > 0) {
+		//we have DO bindings, so we need to shim them
+		const maskedDoBindings = doBindings
+			// Don't shim anything not local to this worker
+			.filter((b) => !b.script_name)
```

Comment:
Doesn't this mean pages DO bindings won't work with D1 🤔

Reply:
@GregBrimble @WalshyDev help

Reply:
I think it's ok, since a Pages worker can't _define_ a DO, it can only connect to one (unless this has changed?). So when you define & deploy the DO itself, this shim will take effect, then Pages will connect to it and Everything'll Be Fine™.

Reply:
Ah okay, that's not an issue then!